File: Part2/Bubble_Sort/bubble_sort.py

input = [5, 3, 2, 4, 1]


def bubble_sort(list_data):
    flag = 1
    count = 0
    while flag:
        flag = 0
        for i in range(0, (len(list_data))):
            if i == 0:
                continue
            if list_data[i] < list_data[i - 1]:
                tmp = list_data[i]
                list_data[i] = list_data[i - 1]
                list_data[i - 1] = tmp
                count += 1
                flag = 1
    return count, list_data


if __name__ == "__main__":
    count, data = bubble_sort(input)
    print(*data)
    print(count)

File: Part2/Bubble_Sort/bubble_sort.c
#include <stdio.h>
#define ARRAY_SIZE 5

int *bubble_sort(int *list)
{
    int i, j, tmp;
    for (i = 0; i < ARRAY_SIZE - 1; i++)
    {
        for (j = i + 1; j < ARRAY_SIZE; j++)
        {
            if (list[j] < list[i])
            {
                tmp = list[i];
                list[i] = list[j];
                list[j] = tmp;
            }
        }
    }
    return list;
}

void main()
{
    int input[ARRAY_SIZE] = {5, 3, 2, 4, 1};

    int *result = bubble_sort(input);

    for (int i = 0; i < ARRAY_SIZE; i++)
    {
        printf("%d ", result[i]);
    }
    printf("\n");
}